modules/javafx.graphics/src/main/native-glass/mac/GlassApplication.m

```diff
@@ -998,13 +1015,17 @@ + (BOOL)syncRenderingDisabled {
 /*
  * Class:     com_sun_glass_ui_mac_MacApplication
  * Method:    _finishTerminating
- * Signature: ()V
+ * Signature: (J)V
  */
 JNIEXPORT void JNICALL Java_com_sun_glass_ui_mac_MacApplication__1finishTerminating
-(JNIEnv *env, jobject japplication)
+(JNIEnv *env, jobject japplication, jlong appDelegate)
 {
     LOG("Java_com_sun_glass_ui_mac_MacApplication__1finishTerminating");
 
+    if (appDelegate) {
+        [(GlassApplication*)appDelegate release];
```

Comment:
This is needed because you moved the allocation of GlassApplication outside (before) the auto-release pool in runLoop, so looks good.